datafusion/core/src/datasource/listing/table.rs

```diff
@@ -256,7 +257,10 @@ impl ListingTable {
     /// If the schema is provided then it must be resolved before creating the table
     /// and should contain the fields of the file without the table
     /// partitioning columns.
-    pub fn try_new(config: ListingTableConfig) -> Result<Self> {
+    pub fn try_new(
+        config: ListingTableConfig,
+        definition: Option<String>,
```

Comment:
This is the only thing I would prefer to change in this PR. Specifically, rather than change the API of `try_new` I think we could add a function like

```rust
pub fn with_definition(mut self, defintion: Option<String>) -> Self {
...
}
```

This change I think  would reduce the size of this PR significantly

Since I realize this PR has been waiting for a week for feedback I will propose this change via PR